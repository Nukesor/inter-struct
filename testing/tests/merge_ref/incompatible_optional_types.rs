use inter_struct::prelude::*;

/// This should produce a compile errors, as the field types for `other` are different.
#[derive(InterStruct, Clone)]
#[merge_ref("crate::MergeStruct")]
pub struct FromStruct {
    pub normal: Option<i32>,
    pub optional: Option<Option<i32>>,
    pub optional_optional: Option<Option<String>>,
}

pub struct MergeStruct {
    pub normal: String,
    pub optional: Option<String>,
    pub optional_optional: Option<Option<String>>,
}

fn main() {}
