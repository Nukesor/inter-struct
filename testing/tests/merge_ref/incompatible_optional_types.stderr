error: Type 'i32 cannot be merged into field of type 'String'.
 --> tests/merge_ref/incompatible_optional_types.rs:7:24
  |
7 |     pub normal: Option<i32>,
  |                        ^^^

error: Type 'Option < Option < i32 > > cannot be merged into field of type 'String'.
 --> tests/merge_ref/incompatible_optional_types.rs:8:19
  |
8 |     pub optional: Option<Option<i32>>,
  |                   ^^^^^^
